### PR TITLE
Add language specifications as markdown file

### PR DESCRIPTION
**Pull request by ThomasMeschke**
In the process of starting to create a lox implementation in C-Sharp in a private repo, I took the time to extract chapter three "The Lox Language" into a standalone markdown file and wanted to contribute this back. Maybe it is of some use.

### CODE REVIEW

File: LOX_LANGUAGE_SPECIFICATIONS.md [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+### Strings
+String literals are enclosed in double quotes. 
+```
+"I am a string";
+"";     // The empty string.
+"123";  // This is a string, not a number.
+```
+
+### Nil
+Used to represend "no value".
```
Comment (kandersen) [inline LOX_LANGUAGE_SPECIFICATIONS.md:56]: represent/represend

Comment (ThomasMeschke) [reply LOX_LANGUAGE_SPECIFICATIONS.md]: 👍

**Comment by aggsol** [comment]
There should be a way to extract it automagicly as some Errata/re-wording might come with the print release in the future and having two places to check could be tedious.

**Comment by munificent** [comment]
Hi, I really appreciate this, but the book is giant enough and I've been working on it long enough that I have to start making hard decisions about what to leave out if I'm ever going to get the print edition done. In this case, I think there is too much overlap with the "The Lox Language" chapter for it to be worth having a separate textual specification of the language. Giving readers two somewhat different explanations of the language will probably just add more confusion than it clarifies.

**Pull request closed without merging**